kernel/packages/shared/apis/SceneStateStorageController/SceneStateStorageController.ts

```diff
@@ -0,0 +1,180 @@
+import { exposeMethod, registerAPI } from 'decentraland-rpc/lib/host'
+import { getFromLocalStorage, saveToLocalStorage } from 'atomicHelpers/localStorage'
+import { ContentClient, DeploymentBuilder } from 'dcl-catalyst-client'
+import { EntityType, Pointer, ContentFileHash } from 'dcl-catalyst-commons'
+import { Authenticator } from 'dcl-crypto'
+import { ExposableAPI } from '../ExposableAPI'
+import { defaultLogger } from '../../logger'
+import { DEBUG } from '../../../config'
+import { CONTENT_PATH, SerializedSceneState } from './types'
+import { getCurrentIdentity } from 'shared/session/selectors'
+import { Asset, AssetId, AssetManager } from './AssetManager'
+import {
+  fromSerializedStateToStorableFormat,
+  fromStorableFormatToSerializedState,
+  StorableSceneState
+} from './StorableSceneStateTranslation'
+import { CLASS_ID } from 'decentraland-ecs/src'
+import { ParcelIdentity } from '../ParcelIdentity'
+import { Store } from 'redux'
+import { RootState } from 'shared/store/rootTypes'
+import { getUpdateProfileServer } from 'shared/dao/selectors'
+import { createGameFile } from './SceneStateDefinitionCodeGenerator'
+
+declare const window: any
+
+@registerAPI('SceneStateStorageController')
+export class SceneStateStorageController extends ExposableAPI {
```

Comment:
We may eventually want to move all this logic to another place in order to decouple this for the SDK.

I'm actually just guessing that we are going to need that, but nonetheless I wanted to plant the idea haha.